### PR TITLE
zsh-forgit: 22.11.0 -> 22.12.0

### PR DESCRIPTION
**Pull request by deejayem**
###### Description of changes

Update zsh-forgit to the latest version.

Previously this was just a single file, but now it calls a helper script, rather than calling fzf and git directly. The helper script sets "SHELL=/bin/bash", so we have to replace that, but for git and fzf I used wrapProgram, to avoid breaking strings like "forgit" which can be displayed to the user.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/shells/zsh/zsh-forgit/default.nix [hunk 1]
```diff
@@ -1,28 +1,44 @@
-{ stdenv, lib, fetchFromGitHub, git, fzf }:
+{ stdenv, lib, pkgs, bash, fetchFromGitHub, makeWrapper }:
```
Comment (SuperSandro2000) [inline pkgs/shells/zsh/zsh-forgit/default.nix:1]:
```suggestion
{ stdenv, lib, bash, fetchFromGitHub, makeWrapper, fzf, git }:
```

File: pkgs/shells/zsh/zsh-forgit/default.nix [hunk 2]
```diff
@@ -1,28 +1,44 @@
...
+
+    runHook postInstall
+  '';
+
+  postInstall = ''
+    wrapProgram $out/bin/git-forgit \
+      --prefix PATH : ${with pkgs; lib.makeBinPath (
+        [fzf git]
+      )}
   '';
```
Comment (SuperSandro2000) [inline pkgs/shells/zsh/zsh-forgit/default.nix:42]:
```suggestion
    wrapProgram $out/bin/git-forgit \
      --prefix PATH : ${lib.makeBinPath [ fzf git ]}

    runHook postInstall
  '';
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by deejayem** [comment]
@offborg build zsh-forgit

**Comment by deejayem** [comment]
@ofborg build zsh-forgit

**Pull request merged**